src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -467,6 +527,26 @@ private RecommendedSymbols GetSymbolsOffOfName(NameSyntax name)
             return new RecommendedSymbols(symbols);
         }
 
+        private RecommendedSymbols GetSymbolsOffOfExpressionInConstantPattern(ExpressionSyntax? originalExpression)
+        {
+            if (originalExpression is null)
+                return default;
+
+            var boundSymbol = _context.SemanticModel.GetSymbolInfo(originalExpression, _cancellationToken).Symbol;
+
+            if (boundSymbol is not INamespaceOrTypeSymbol namespaceOrType)
+                return default;
+
+            // The RHS of an `is` pattern may only include qualifications to
+            // - namespaces (from other namespaces or aliases),
+            // - types (from aliases, namespaces or other types),
+            // - constant fields (from types)
+            // Methods, proprties, events, non-constant fields etc. are excluded since they may not be present in the RHS of an `is` pattern
```

Comment:
what about something like `if (x is nameof(A.SomeProperty)` ?

Reply:
Added passing test